src/reader-activation-auth/auth-modal.js

```diff
@@ -22,34 +22,34 @@ export function openAuthModal( config = {} ) {
 	const modalTrigger = config.trigger;
 
 	if ( reader?.authenticated ) {
-		if ( config.callback ) {
-			config.callback();
+		if ( config.onSuccess && typeof config.onSuccess === 'function' ) {
+			config.onSuccess();
```

Comment:
Non-blocking suggestion: You can remove the need for these checks throughout the PR (`config.onSuccess && ... === 'function'`) by just calling `config?.onSuccess?.()`

Reply:
This will still throw an error if `onSuccess` is truthy

Reply:
Hmm, this bit should cause this to resolve as undefined when not a function: `onSuccess?.()`

Reply:
I double-checked running `const notAFn = true; notAFn?.();` in the console and it errored.

Reply:
Right. This won't work on every browser, but our build process will be sure to compile this to something universally compatible. This isn't a blocker though so feel free to keep what you have!

Reply:
For reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

Reply:
I'd rather keep it because the optional chaining operator will not do type validation.